File: IoTEx.WaternetIoT/IoTEx.WaternetIoT.Model/DTOs/DeviceDefinitionDTO.cs


namespace IoTEx.WaternetIoT.Model.DTOs
{

    public class DeviceDefinitionDTO
    {

        public string id { get; set; }
        public string deviceId { get; set; }
        public string deviceName { get; set; }
        public string assetUID { get; set; }
        public bool isTraced { get; set; }
        public bool isActive { get; set; }
        public DeviceDefinitionInfoDTO info { get; set; }
        public DeviceDefinitionLocationDTO location { get; set; }        
        public DeviceDefinitionNetworkInfoDTO network { get; set; }
        public List<DeviceDefinitionProjectDTO> projects { get; set; }
        public DateTime publishedDate { get; set; }
        public DateTime? installedDate { get; set; }
        public int publishedCounter { get; set; }
        public string publishedByUserId { get; set; }
        public string publishedByUsername { get; set; }
        public DeviceDefinitionSettingsDTO settings  { get; set; }

    }
    
    
    
    
    
}

File: IoTEx.WaternetIoT/IoTEx.Waternet.Portal/src/app/models/DeviceDefinitionDTO.ts
import { DeviceInfoDTO, DeviceLocationDTO, DeviceNetworkInformationDTO, DeviceProjectDTO, DeviceDefinitionSettingsDTO } from "./_index";

export class DeviceDefinitionDTO {
  id?: string;
  deviceId?: string;
  deviceName?: string;
  assetUID?: string;
  isTraced?: boolean = false;
  isActive?: boolean = false;
  info?: DeviceInfoDTO;
  location?: DeviceLocationDTO;
  network?: DeviceNetworkInformationDTO;
  projects?: DeviceProjectDTO[];
  publishedDate?: Date;
  installedDate?: Date;
  publishedCounter?: number = 0;
  publishedByUserId?: string;
  publishedByUsername?: string;
  settings?: DeviceDefinitionSettingsDTO;
}
